File: solutions/go/hamming/2/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
    runeA, runeB := []rune(a), []rune(b)
	if len(runeA) != len(runeB) {
        return -1, errors.New("Two strands must have equal length")
    }
    dist := 0
    for i, v := range runeA {
        if runeB[i] != v {
            dist++
        }
    }
    return dist, nil
}

File: solutions/lua/hamming/2/hamming.lua
local function compute(a, b)
  if string.len(a) ~= string.len(b) then
    error("strands must be of equal length")
  end
  local acc = 0
  for i = 1, string.len(a) do
    if a:sub(i, i) ~= b:sub(i, i) then
      acc = acc + 1
    end
  end
  return acc
end

return {compute = compute}